engine/development/deploymentManager/src/main/scala/pl/touk/nussknacker/development/manager/DevelopmentDeploymentManagerProvider.scala

```diff
@@ -121,28 +121,27 @@ class DevelopmentDeploymentManager(actorSystem: ActorSystem)
 
   override def customActions: List[CustomAction] = customActionStatusMapping.keys.toList
 
-  override def invokeCustomAction(actionRequest: CustomActionRequest, canonicalProcess: CanonicalProcess): Future[Either[CustomActionError, CustomActionResult]] =
-    Future.successful(
-      customActions
-        .find(_.name.equals(actionRequest.name))
-        .map(customAction => {
-          val processName = actionRequest.processVersion.processName
-          val statusDetails = memory.getOrElse(processName, createAndSaveProcessState(NotDeployed, actionRequest.processVersion))
-
-          if (customAction.allowedStateStatusNames.contains(statusDetails.status.name)) {
-            customActionStatusMapping
-              .get(customAction)
-              .map { status =>
-                asyncChangeState(processName, status)
-                Right(CustomActionResult(actionRequest, s"Done ${actionRequest.name}"))
-              }
-              .getOrElse(Left(CustomActionInvalidStatus(actionRequest, statusDetails.status.name)))
-          } else {
-            Left(CustomActionInvalidStatus(actionRequest, statusDetails.status.name))
-          }
-        })
-        .getOrElse(Left(CustomActionNotImplemented(actionRequest)))
-    )
+  override def invokeCustomAction(actionRequest: CustomActionRequest, canonicalProcess: CanonicalProcess): Future[CustomActionResult] =
+    customActions
+      .find(_.name.equals(actionRequest.name))
+      .map(customAction => {
+        val processName = actionRequest.processVersion.processName
+        val statusDetails = memory.getOrElse(processName, createAndSaveProcessState(NotDeployed, actionRequest.processVersion))
+
+        if (customAction.allowedStateStatusNames.contains(statusDetails.status.name)) {
+          customActionStatusMapping
+            .get(customAction)
+            .map { status =>
+              asyncChangeState(processName, status)
+              Future.successful(CustomActionResult(actionRequest, s"Done ${actionRequest.name}"))
+            }
+            .getOrElse(Future.failed(CustomActionInvalidStatus(actionRequest, statusDetails.status.name)))
```

Comment:
IMO this example need refactor. I don't understand why someone would want to check `customAction.allowedStateStatusNames.contains(statusDetails.status.name)` by himself inside DeploymnetManager. It is already checked inside `CustomActionInvokerService`. I think that we should remove all these `CustomActionError` implementations from `extensions-api` module and make them internal (`designer` module)